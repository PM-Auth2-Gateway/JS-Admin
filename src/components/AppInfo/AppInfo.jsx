import React from 'react';
import { Link, useRouteMatch } from 'react-router-dom';

import {
  Container,
  Row,
  Col,
  FormGroup,
  FormLabel,
  FormControl,
  Button,
  ListGroup,
  ListGroupItem,
  Card,
  Alert,
} from 'react-bootstrap';

import styles from './AppInfo.module.scss';

import appIcon from '../../assets/app-icon.svg';
import googleIcon from '../../assets/google-avatar.png';
import facebookIcon from '../../assets/facebook-avatar.png';

const AppInfo = () => {
  const { url } = useRouteMatch();

  return (
    <Container>
      <div className={styles.appPreview}>
        <img src={appIcon} className={styles.appIcon} alt="icon" />
        <h2>App name</h2>
        {/*TODO from backend*/}
      </div>
      <div className={styles.form}>
        <ListGroup variant={'flush'}>
          <ListGroupItem>
            <Row>
              <Col>
                <h4>Basic Information</h4>
              </Col>
              <Col>
                {/*TODO get from backend basic info*/}
                <FormGroup>
                  <FormLabel>Name</FormLabel>
                  <FormControl type={'text'} />
                </FormGroup>
                <FormGroup>
                  <FormLabel>App ID</FormLabel>
                  <FormControl
                    type={'text'}
                    readOnly
                    defaultValue={'sadasdsadsad'}
                  />
                </FormGroup>

                <Button variant={'info'}>Save Changes</Button>
              </Col>
            </Row>
          </ListGroupItem>
          <ListGroupItem>
            <Row>
              <Col>
                <h4>Social networks</h4>
              </Col>
              <Col>
                {/*TODO get from backend social network list*/}
                <Card className={styles.card}>
                  <Link to={`${url}/social/1`}>
                    <Card.Body>
                      <Card.Title className={styles.socialPreview}>
                        <img
                          src={googleIcon}
                          className={styles.appIcon}
                          alt="icon"
                        />
                        <p>Google</p>
                      </Card.Title>
                      <Card.Text className={'text-muted'}>
                        Allow your users to login with their Google Account
                      </Card.Text>
                    </Card.Body>
                  </Link>
                </Card>
                <Card className={styles.card}>
                  <Link to={`${url}/social/2`}>
                    <Card.Body>
                      <Card.Title className={styles.socialPreview}>
                        <img
                          src={facebookIcon}
                          className={styles.appIcon}
                          alt="icon"
                        />
                        <p>Facebook</p>
                      </Card.Title>
                      <Card.Text className={'text-muted'}>
                        A fast and convenient way for users to log into your app
                        with Facebook
                      </Card.Text>
                    </Card.Body>
                  </Link>
                </Card>
              </Col>
            </Row>
          </ListGroupItem>
        </ListGroup>
      </div>

      <h3 className={'my-3'}>Danger zone</h3>
      <Alert variant={'danger'}>
        <Row>
          <Col>
            <h5>Delete this application</h5>
            <p>All your apps using this client will stop working.</p>
          </Col>
          <Col xs="auto" className={'align-self-center'}>
            <Button variant={'danger'}>Delete</Button>
          </Col>
        </Row>
      </Alert>
    </Container>
  );
};

export default AppInfo;
