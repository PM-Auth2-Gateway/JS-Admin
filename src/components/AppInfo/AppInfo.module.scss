.appPreview {
  display: flex;
  align-items: flex-start;
  gap: 24px;
  margin-bottom: 48px;

  .appIcon {
    width: 64px;
    border-radius: 6px;
    background-color: #eff0f2;
  }
}

.form {
  border: 1px solid #c9cace;
  padding: 32px;
  border-radius: 6px;
}

.card {
  margin-bottom: 20px;

  a {
    text-decoration: none;
  }

  .socialPreview {
    display: flex;
    align-items: flex-start;
    gap: 16px;

    .appIcon {
      width: 32px;
      border-radius: 50%;
    }
  }
}



