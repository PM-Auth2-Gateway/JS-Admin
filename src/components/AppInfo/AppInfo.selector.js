import { createStructuredSelector } from 'reselect';

const currentSelector = (globalState) => globalState.appsCurrent.value;
const loadingSelector = (globalState) => globalState.appsCurrent.loading;
const errorSelector = (globalState) => globalState.appsCurrent.error;

const selector = createStructuredSelector({
  current: currentSelector,
  loading: loadingSelector,
  error: errorSelector,
});

export default selector;
