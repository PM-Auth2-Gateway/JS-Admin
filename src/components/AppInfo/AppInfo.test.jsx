import React from 'react';
import { shallow } from 'enzyme';
import { useRouteMatch } from 'react-router-dom';

import AppInfo from './AppInfo';

jest.mock('react-router-dom', () => ({
  useRouteMatch: jest.fn(),
  Link: 'Link',
}));

describe('AppInfo Component', () => {
  it('should render AppInfo Component', function () {
    useRouteMatch.mockReturnValue({ url: '/applications/1' });
    const component = shallow(<AppInfo />);

    expect(component).toMatchSnapshot();
  });
});
