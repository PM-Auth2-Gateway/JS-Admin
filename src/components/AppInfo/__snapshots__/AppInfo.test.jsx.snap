// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppInfo Component should render AppInfo Component 1`] = `
<Container
  fluid={false}
>
  <div
    className="appPreview"
  >
    <img
      alt="icon"
      className="appIcon"
      src="app-icon.svg"
    />
    <h2>
      App name
    </h2>
  </div>
  <div
    className="form"
  >
    <ListGroup
      variant="flush"
    >
      <ListGroupItem
        active={false}
        disabled={false}
      >
        <Row
          noGutters={false}
        >
          <Col>
            <h4>
              Basic Information
            </h4>
          </Col>
          <Col>
            <FormGroup>
              <FormLabel
                column={false}
                srOnly={false}
              >
                Name
              </FormLabel>
              <FormControl
                type="text"
              />
            </FormGroup>
            <FormGroup>
              <FormLabel
                column={false}
                srOnly={false}
              >
                App ID
              </FormLabel>
              <FormControl
                defaultValue="sadasdsadsad"
                readOnly={true}
                type="text"
              />
            </FormGroup>
            <Button
              active={false}
              disabled={false}
              variant="info"
            >
              Save Changes
            </Button>
          </Col>
        </Row>
      </ListGroupItem>
      <ListGroupItem
        active={false}
        disabled={false}
      >
        <Row
          noGutters={false}
        >
          <Col>
            <h4>
              Social networks
            </h4>
          </Col>
          <Col>
            <Card
              body={false}
              className="card"
            >
              <Link
                to="/applications/1/social/1"
              >
                <CardBody>
                  <CardTitle
                    className="socialPreview"
                  >
                    <img
                      alt="icon"
                      className="appIcon"
                      src="google-avatar.png"
                    />
                    <p>
                      Google
                    </p>
                  </CardTitle>
                  <CardText
                    className="text-muted"
                  >
                    Allow your users to login with their Google Account
                  </CardText>
                </CardBody>
              </Link>
            </Card>
            <Card
              body={false}
              className="card"
            >
              <Link
                to="/applications/1/social/2"
              >
                <CardBody>
                  <CardTitle
                    className="socialPreview"
                  >
                    <img
                      alt="icon"
                      className="appIcon"
                      src="facebook-avatar.png"
                    />
                    <p>
                      Facebook
                    </p>
                  </CardTitle>
                  <CardText
                    className="text-muted"
                  >
                    A fast and convenient way for users to log into your app with Facebook
                  </CardText>
                </CardBody>
              </Link>
            </Card>
          </Col>
        </Row>
      </ListGroupItem>
    </ListGroup>
  </div>
  <h3
    className="my-3"
  >
    Danger zone
  </h3>
  <Alert
    closeLabel="Close alert"
    show={true}
    transition={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "appear": false,
          "in": false,
          "mountOnEnter": false,
          "timeout": 300,
          "unmountOnExit": false,
        },
        "render": [Function],
      }
    }
    variant="danger"
  >
    <Row
      noGutters={false}
    >
      <Col>
        <h5>
          Delete this application
        </h5>
        <p>
          All your apps using this client will stop working.
        </p>
      </Col>
      <Col
        className="align-self-center"
        xs="auto"
      >
        <Button
          active={false}
          disabled={false}
          variant="danger"
        >
          Delete
        </Button>
      </Col>
    </Row>
  </Alert>
</Container>
`;
